stdlib/universe/reduce.go

```diff
@@ -151,60 +162,112 @@ func (t *reduceTransformation) Process(id execute.DatasetID, tbl flux.Table) err
 				return errors.Wrap(err, codes.Inherit, "failed to evaluate reduce function")
 			}
 			params[accumulatorParamName] = m
+
+			key := t.computeGroupKey(tbl.Key(), m)
```

Comment:
This seems entirely duplicated from the same code below. Should factor it out to a function if it does not make sense to keep it in the loop (as the added comment mentions).